sgx_urts/src/file.rs

```diff
@@ -402,31 +414,35 @@ pub extern "C" fn u_symlink_ocall(
 pub extern "C" fn u_realpath_ocall(
     error: *mut c_int,
     pathname: *const c_char,
-) -> *mut c_char {
+    resolved_buf: *mut c_char,
+    _bufsz: size_t,
+) -> c_int {
     let mut errno = 0;
-    let ret = unsafe { libc::realpath(pathname, ptr::null_mut()) };
+    let mut result = 0;
+    let ret = unsafe { libc::realpath(pathname, resolved_buf) };
     if ret.is_null() {
         errno = Error::last_os_error().raw_os_error().unwrap_or(0);
+        result = -1;
```

Comment:
`libc::realpath` return `NULL` on error. should we do this as well?

Reply:
This is another case shows that our new shim layer API differs from the one in libc. The new shim API provide similar functionality with a less error-prone interface. 
The new `u_realpath_ocall` shim API force the caller to provide a buffer to hold the resolved path information, and returns a `c_int` (instead of a useless `*mut c_char`) to indicates success or error.